common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -661,18 +683,47 @@
                 this.showLanguageMenu = true;
 
                 $.each(languages, function(code, label) {
-                    var li = $('<li data-lang-code="' + code + '" />'),
-                        link = $('<button class="control control-lang">' + label + '</button>');
+                    li = HtmlUtils.interpolateHtml(
+                        HtmlUtils.HTML(
+                            '<li data-lang-code="{code}" />'
+                        ),
+                        {
+                            code: code
+                        }
+                    ).toString();
+                    
+                    link = HtmlUtils.interpolateHtml(
+                        HtmlUtils.HTML(
+                            '<button class="control control-lang" aria-pressed="false">{label}</button>'
+                        ),
+                        {
+                            label: label
+                        }
+                    ).toString();
+                    
+                    liObj = $.parseHTML(li);
```

Comment:
Agreed. There's no benefit to converting to an HTML snippet only to turn it back into a string. IMO we should also never use ``parseHTML`` as that is just as dangerous as ``$`` (I don't know if we lint against it currently, but it seems we should).

The way to make the linter happy is to use the naming convention that all JQuery elements should be stored in variables starting with ``$``. So, use ``$menu``, ``$li``, ``$link`` etc. Once you have an element, you can safely append them without using ``HtmlUtils.append``.

I think something like this should work (UNTESTED!):

    var $li = $('<li />').data('lang-code', code),
          linkHtml = HtmlUtils.joinHtml(
               HtmlUtils.HTML('<button class="control control-lang">'),
               label,
               HtmlUtils.HTML('</button>')
          ),
          $link = $(linkHtml.toString());
    if (currentLang === code) {
        $li.addClass('is-active');
        $link.attr('aria-pressed', 'true');
     }
     $li.append($link);
     $menu.append($li);

This would all be simpler written as an Underscore template, but that may be too big a refactoring.

Let's talk in person if this doesn't make sense.